taskfiles/utils.yml

```diff
@@ -141,3 +146,159 @@ tasks:
         \( -iname "*.cpp" -o -iname "*.h" -o -iname "*.hpp" \) \
         -print0 | \
           xargs -0 clang-tidy {{.FLAGS}}
+
+  # ===
+  # CMAKE UTILS
+  # ===
+
+  # Runs cmake configure and build steps for the given source and build directories.
+  #
+  # @param {string} BUILD_DIR Cmake build directory to create.
+  # @param {string} SOURCE_DIR Project source directory containing the CMakeLists.txt file.
+  # @param {string}[optional] CHECKSUM_FILE Path to store the checksum of built files.
+  # @param {string}[optional] CMAKE_ARGS Any additional arguments to pass to cmake configure.
+  cmake-build:
+    label: "cmake-build: {{.SOURCE_DIR}} {{.BUILD_DIR}}"
+    internal: true
+    vars:
+      CHECKSUM_FILE: '{{default (printf "%s.md5" .BUILD_DIR) .CHECKSUM_FILE}}'
+      CMAKE_ARGS: "{{default nil .CMAKE_ARGS}}"
```

Comment:
Would it be better to fallback to `""`?

Reply:
Sure we can. I don't know if it makes a difference, but I don't think it can hurt either.